atomix/cluster/src/test/java/io/atomix/raft/RaftTest.java

```diff
@@ -1178,444 +485,24 @@ public void testThreeNodesSequentiallyStart() throws Throwable {
-              resume();
-            });
-
-    for (int i = 0; i < 10; i++) {
-      primitive.sendEvent(false).thenRun(this::resume);
-      await(10000, 4);
-    }
-  }
-
-  /** Tests session expiring events. */
-  @Test
-  public void testOneNodeExpireEvent() throws Throwable {
-    testSessionExpire(1);
-  }
-
-  /** Tests a session expiring. */
-  private void testSessionExpire(final int nodes) throws Throwable {
-    createServers(nodes);
-
-    final RaftClient client1 = createClient();
-    final TestPrimitive primitive1 = createPrimitive(client1);
-    final RaftClient client2 = createClient();
-    createSession(client2);
-    primitive1.onExpire(event -> resume()).thenRun(this::resume);
-    client2.close().thenRun(this::resume);
-    await(Duration.ofSeconds(10).toMillis(), 3);
-  }
-
-  /** Tests session expiring events. */
-  @Test
-  public void testThreeNodeExpireEvent() throws Throwable {
-    testSessionExpire(3);
-  }
-
-  /** Tests session expiring events. */
-  @Test
-  public void testFiveNodeExpireEvent() throws Throwable {
-    testSessionExpire(5);
-  }
-
-  /** Tests session close events. */
-  @Test
-  public void testOneNodeCloseEvent() throws Throwable {
-    testSessionClose(1);
-  }
-
-  /** Tests a session closing. */
-  private void testSessionClose(final int nodes) throws Throwable {
-    createServers(nodes);
-
-    final RaftClient client1 = createClient();
-    final TestPrimitive primitive1 = createPrimitive(client1);
-    final RaftClient client2 = createClient();
-    primitive1.onClose(event -> resume()).thenRun(this::resume);
-    await(Duration.ofSeconds(10).toMillis(), 1);
-    createSession(client2).close().thenRun(this::resume);
-    await(Duration.ofSeconds(10).toMillis(), 2);
-  }
-
-  /** Tests session close events. */
-  @Test
-  public void testThreeNodeCloseEvent() throws Throwable {
-    testSessionClose(3);
-  }
-
-  /** Tests session close events. */
-  @Test
-  public void testFiveNodeCloseEvent() throws Throwable {
-    testSessionClose(5);
-  }
-
   @Test
   public void testThreeNodeManyEventsDoNotMissHeartbeats() throws Throwable {
     // given
     createServers(3);
+    final var leader = getLeader(servers).orElseThrow();
 
-    final RaftClient client = createClient();
-    final TestPrimitive primitive = createPrimitive(client);
-    primitive.onEvent(
-        message -> {
-          threadAssertNotNull(message);
-          resume();
-        });
+    appendEntry(leader);
 
     final double startMissedHeartBeats = RaftRoleMetrics.getHeartbeatMissCount("1");
 
     // when
-    for (int i = 0; i < 1_000; i++) {
-      primitive.sendEvent(true);
-    }
-    await(10000, 1_000);
+    appendEntries(leader, 1000);
 
     // then
     final double missedHeartBeats = RaftRoleMetrics.getHeartbeatMissCount("1");
     assertThat(0.0, is(missedHeartBeats - startMissedHeartBeats));
   }
 
-  @Test
-  public void testSnapshotSentOnDataLoss() throws Throwable {
```

Comment:
Did you move this test to some where else? This is important to test.

Reply:
As I explained, this tests doesn't make sense with the ZeebeRaftStateMachine. This test can't be done in the Raft module, since we are not able to create normal snapshot here. I created a follow up issue #4468 to investigate them. Maybe it is possible to create fake snapshots otherwise we have to test it elsewhere. Furthermore I thought that we have such a test already in QA.